### ISSUE TITLE
[BUG] Timestamp::now() exposed via hdk, but always panics

### ISSUE THREAD

**Issue by mattyg** [holochain/holochain#3003]
**Describe the bug**
~I get a panic when calling `Timestamp::now()` on `develop-0.2`~

~(There's a chance this may not be a bug, but instead related to running the tests via holonix version `0_2` while my holochain dev-dependency is from git branch `develop-0.2` -- if you think that's the case let me know)~

~If it is a bug this is high priority for me to fix & have tests in place before 0.2.3.~

I'm able to call `Timestamp::now()` via hdk, but it will always panic.

**Expected behavior**
~Can get current timestamp from host~

`Timestamp::now()` shouldn't be available via hdk / hdi

**System information:**
 - OS: Ubuntu 22.04
 - Holochain and HDK Version: holonix version is 0.2.2, but sweetests, hdk, and hdi are pulling from git `develop-0.2` branch

**Steps to reproduce**

minimal reproduction: https://github.com/mattyg/bug-reproduction-timestamp

`npm run build:happ && cargo test`

**Comment by mattyg** [comment]
It looks like the feature `full` from `kitsune_p2p_timestamp` needs to get bubbled up to the holochain crate and set as part of it's own `test_utils` feature

**Comment by jost-s** [comment]
Hi @mattyg, yeah, I've noticed this doesn't work too. Thanks for the reproduction, it's reproducible.

According to the `Timestamp` docs it's not supposed to work in WASMs https://github.com/holochain/holochain/blob/develop/crates/kitsune_p2p/timestamp/src/lib.rs#L46-L47

I don't know what's the recommended way here, but you can use sys_time as well as `chrono::prelude::Utc::now().timestamp()`.

**Comment by mattyg** [comment]
I see, then yeah it seems like there is just something funky with the feature flags being passed into kitsune_p2p_timestamp when running holochain feature test_utils. Likely helped by this absurd feature naming :laughing: 

![image](https://github.com/holochain/holochain/assets/159270/fa5cae59-1ac9-47fc-a0ae-37b1e77d446d)

**Comment by nphias** [comment]
i found that chrono::prelude::Utc::now() also causes a runtime error. (hc 0.2.3-rc.1)

**Comment by mattyg** [comment]
> i found that chrono::prelude::Utc::now() also causes a runtime error. (hc 0.2.3-rc.1)

Yeah we can't get chrono timestamps in the wasm guest (need to ask the host for timestamp) -- that's also why Timestamp::now() panics. We have a function sys_time() for use within the wasm guest.

I think the only issue here is that Timestamp::now() either shouldn't be exposed via the hdk, or should have a feature flag to switch its implementation to use sys_time() when compiled for wasm targets

**Comment by ThetaSinner** [comment]
I've got a fix for this. It is feature indeed an unhelpful combination of features through dependency crates. I found a reasonably nice solution in `holochain_zome_types` that resolves the issue, with a few other tidy-ups.

I even managed to make a test which ensures this stays unavailable which I think is pretty cool, because I get excited by testing tools / libraries that do that kind of thing :D
